spec/Variables.tex

```diff
@@ -414,11 +417,97 @@ \section{Configuration Variables}
 \begin{chapeloutput}
 2
 \end{chapeloutput}
-sets a integer parameter \chpl{rank} to \chpl{2}.  At
-compile-time, \chpl{rank} can be set via a configuration file or compile-line
-override to \chpl{3} or \chpl{2*n}\footnote{It is assumed here that \chpl{n} is
-also a param variable.} or indeed to any other expression
-that can be evaluated at compile-time.  The value supplied at compile time
-overrides the value \chpl{2} appearing in the code.  In this example, the \chpl{rank}
+sets an integer parameter \chpl{rank} to \chpl{2}.
+At compile-time, this default value of \chpl{rank} can be overridden
+with another parameter expression, such as \chpl{3} or \chpl{2*n},
+provided \chpl{n} itself is a parameter. The \chpl{rank}
 configuration variable can be used to write rank-independent code.
 \end{chapelexample}
+
+\section{Ref Variables}
+\label{Ref_Variables}
+\index{variables!ref}
+\index{ref@\chpl{ref}}
+
+A \emph{ref} variable is a variable declared using the \chpl{ref} keyword.
+A ref variable serves as an alias to another variable.
+% TODO: once the bug is fixed, add: another variable, field or array element.
+The declaration of a ref variable must contain \sntx{initialization-part},
+which specifies what is to be aliased and can be a variable
+or any lvalue expression.
+
+Access or update to a ref variable is equivalent to access or update
+to the variable being aliased, subject to memory consistency model
+\rsec{Memory_Consistency_Model}. For example, an update to a ref variable
+is visible via the original variable, and visa versa.
+
+If the variable being aliased is a runtime constant, the corresponding
```

Comment:
Right, but there are other ways to have a const expression (e.g. return by const ref intent, argument passed in by const ref). In these situations, it doesn't matter if the variable being aliased is declared with const or var.